large_page-c/large_page.c

```diff
@@ -287,11 +287,11 @@ static void AlignRegionToPageBoundary(mem_range* r) {
 }
 
 static map_status CheckMemRange(mem_range* r) {
-  if (r->from == NULL || r->to == NULL || r->from > r->to) {
+  if (r->from == NULL || r->to == NULL) {
     return map_invalid_region_address;
   }
 
-  if (r->to - r->from < HPS) {
+  if (r->to - r->from < HPS || r->from > r->to) {
```

Comment:
Does this condition happen where from > to? We were treating it as an invalid region

Reply:
"r->from > r->to" condition occurs for smaller regions after AlignRegionToPageBoundary is called ('from' is aligned up and 'to' is aligned down). This part of commit is not critical. I can drop it if needed.

Reply:
before AlignRegionToPageBoundary : from < to

Reply:
Thanks, that makes sense.